avro-builder/src/main/java/com/linkedin/avroutil1/builder/operations/codegen/own/AvroUtilCodeGenOp.java

```diff
@@ -46,38 +49,42 @@ public AvroUtilCodeGenOp(CodeGenOpConfig config) {
 
   @Override
   public void run(OperationContext opContext) throws Exception {
-    //mkdir any output folders that dont exist
+    //mkdir any output folders that don't exist
     if (!config.getOutputSpecificRecordClassesRoot().exists() && !config.getOutputSpecificRecordClassesRoot()
         .mkdirs()) {
       throw new IllegalStateException(
           "unable to create destination folder " + config.getOutputSpecificRecordClassesRoot());
     }
 
-    List<Path> avscFiles = new ArrayList<>();
+    Set<File> avscFiles = new HashSet<>();
+    Set<File> importableFiles = new HashSet<>();
+    String[] extensions = new String[]{BuilderConsts.AVSC_EXTENSION};
+
+
     for (File inputRoot : config.getInputRoots()) {
```

Comment:
i think input should be the importable one, and the non-importable sources should get another name.
long term we want to have all inputs be importable, and eliminate usage of non-importable sources